src/wallet/scriptpubkeyman.cpp

```diff
@@ -53,7 +53,7 @@ bool ScriptPubKeyMan::CanGenerateKeys()
 {
     // A wallet can generate keys if it has an HD seed (IsHDEnabled) or it is a non-HD wallet (pre FEATURE_HD)
     LOCK(wallet->cs_wallet);
-    return IsHDEnabled();
+    return IsHDEnabled() || wallet->GetVersion() < FEATURE_PRE_SPLIT_KEYPOOL;
```

Comment:
I'm not seeing a point here, this can never be false.

Reply:
Without this check, this would be false on non-HD wallets, as it can be easily verified now running the tests with `--legacywallet`.